### ISSUE TITLE
Feature request: JSONC support

### ISSUE THREAD

**Issue by TrevorBurnham** [amzn/style-dictionary#698]
Currently, style-dictionary supports [JSON5](https://json5.org/) for token files (PR: https://github.com/amzn/style-dictionary/pull/165). JSON5 is used in the [transitive-transforms example](https://github.com/amzn/style-dictionary/tree/main/examples/advanced/transitive-transforms) "so that it can be marked up with comments for reference." YAML is also supported, and [has its own example project](https://github.com/amzn/style-dictionary/blob/main/examples/advanced/yaml-tokens/).

I think it's great that these alternative formats are supported, but vanilla JSON still seems to be the most popular format for storing tokens. As a developer, it's the format I'm most comfortable with. But, I'd really like to be able to add comments to my token files. There's a format that would make that possible: JSONC, aka "JSON with Comments".

Using JSONC, this would be possible:

```jsonc
{
  "color": {
    "font": {
      // Colors used for text on a light background
      "base": { "value": "#111111" },
      "secondary": { "value": "#333333" },
      "tertiary": { "value": "#666666" },

      // Colors used for text on a dark background
      "inverse": {
        "base": { "value": "#ffffff" }
      }
    }
  }
}
```

The JSONC format is very popular. There's even built-in syntax support in VS Code:

<img width="590" alt="JSON with Comments syntax selection" src="https://user-images.githubusercontent.com/224895/134058017-57e13ef0-134d-48ad-83e8-a7120fe4d7f4.png">

In short, I think JSONC would be a perfect format for tokens. 😄

**Comment by chazzmoney** [comment]
Hi @TrevorBurnham 

Thanks for the suggestion - we would love to support JSONC if there is a straightforward way of doing so. Do you think you could submit a PR for us?

To get you started, [this is where the configuration is actually loaded](https://github.com/amzn/style-dictionary/blob/main/lib/extend.js#L94); we currently just piggyback off of require().

Looking forward to merging in a JSONC configuration PR in the future!

**Comment by TrevorBurnham** [comment]
@chazzmoney Thanks for the encouragement! I've opened a PR: https://github.com/amzn/style-dictionary/pull/732
